### ISSUE TITLE
igraph_betweenness_estimate() returns all-zero result with cutoff=0

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1050]
The documentation of `igraph_betweenness_estimate()` says,

```
 * \param cutoff The maximal length of paths that will be considered.
 *        If zero or negative, the exact betweenness will be calculated
 *        (no upper limit on path lengths).
```

However, with `cutoff=0`, it returns an all-zero vector.  This is true both for weighted and unweighted calculations.

Negative cutoff values do calculate the exact betweenness, as they should.

`igraph_edge_betweenness_estimate()` suffers from the same problem, but only for weighted calculations.

This is a report for the C core (latest git), but here's a demo with R:

```
> g<-erdos.renyi.game(10,20,'gnm')
> betweenness.estimate(g,cutoff=0)
 [1] 0 0 0 0 0 0 0 0 0 0
> betweenness.estimate(g,cutoff=-1)
 [1] 6.0833333 2.9166667 1.2500000 1.9166667 0.7916667 1.9166667 3.2500000 6.8333333 1.2500000
[10] 0.7916667
```

**Comment by szhorvat** [comment]
One of the things I would like to know is if this will be fixed by changing the function to return exact results for `cutoff=0` or changing the documentation to require `cutoff < 0` instead.

**Comment by ntamas** [comment]
`cutoff=0` does not really make sense to me (I mean, why would anyone run the function with `cutoff=0`?), so I think it's okay if we fix it by bending the implementation to the documentation (and not vice versa). Any objections?

**Comment by szhorvat** [comment]
Fixed by #1187
